templates/submission/status.html

```diff
@@ -69,7 +69,9 @@
 
 {% block body %}
     <br>
-    <div><a href="{{ url('submission_source', submission.id) }}">{{ _('View source') }}</a></div>
+    {% if submission.language.file_only and request.user.is_superuser %}
```

Comment:
This makes nobody can view any submission source if the language is not file only

The correct way is: `if not submission.langue.file_only or request.user.is_superuser`